js-fiddle/api.js

```diff
@@ -0,0 +1,388 @@
+  cleanUi();
+  page_editor.style.display = 'block';
+  page_help.style.display = 'none';
+  editor.setModel(lang === 'js' ? jsModel : htmlModel);
+  (lang == 'js' ? btnTabJs : btnTabHtml).style.background = 'green';
+}
+
+function installWidget(code, html) {
+  state('installed');
+  code = code ?? jsModel.getValue();
+  html = html ?? htmlModel.getValue();
+  createFrame();
+  const content = wFrame.contentWindow;
+  content.document.open();
+  content.document.write(html);
+  if (code.trim()) {
+    if (!html.includes('grist-plugin-api.js')) {
+      content.document.write(
+        `<script src="https://docs.getgrist.com/grist-plugin-api.js"></` +
+          `script>`
+      );
+    }
+    content.document.write(`<script>${code}</` + `script>`);
+  }
+  content.document.close();
+  page_widget.style.display = 'block';
+  page_editor.style.display = 'none';
+  page_help.style.display = 'none';
+
+  [...document.getElementsByClassName('_tab')].forEach(
+    e => (e.style.display = 'none')
+  );
+  [...document.getElementsByClassName('_button')].forEach(
+    e => (e.style.display = 'none')
+  );
+}
+
+async function showEditor() {
+  state('editor');
+  await loadMonaco();
+  init();
+
+  bar.style.display = 'flex';
+  page_editor.style.display = 'block';
+  page_help.style.display = 'none';
+  buildEditor();
+  jsModel.setValue(currentJs() ?? DEFAULT_JS);
+  htmlModel.setValue(currentHtml() ?? DEFAULT_HTML);
+
+  page_widget.style.display = 'none';
+  page_editor.style.display = 'block';
+
+  [...document.getElementsByClassName('_tab')].forEach(
+    e => (e.style.display = 'inline-block')
+  );
+  [...document.getElementsByClassName('_button')].forEach(
+    e => (e.style.display = 'none')
+  );
+  btnInstall.style.display = 'inline-block';
+  btnReset.style.display = 'inline-block';
+  changeTab('html');
+}
+
+const onOptions = function (clb) {
+  let listen = true;
+  let last = undefined;
+  grist.onOptions((...data) => {
+    if (listen) {
+      if (last !== undefined) {
+        if (JSON.stringify(last) === JSON.stringify(data)) {
+          return;
+        }
+        last = data;
+      }
+      clb(...data);
+    }
+  });
+  return () => void (listen = false);
+};
+
+onOptions(async options => {
+  if (!options) {
+    if (state() === 'installed') {
+      // If we are already installed, check that we don't have any code shown.
+      if (currentJs() || currentHtml()) {
+        // If we have, remove it and install default widget.
+        currentHtml(null);
+        currentJs(null);
+        installWidget(DEFAULT_JS, DEFAULT_HTML);
+      } else {
+        // Don't need to do anything.
+      }
+    } else {
+      // We are not installed, so install the default widget.
+      currentHtml(null);
+      currentJs(null);
+      state('installed');
+      installWidget(DEFAULT_JS, DEFAULT_HTML);
+    }
+  } else {
+    // Install the widget from options, if we don't have any code in memory.
+    const newJs = options?._js ?? null;
```

Comment:
Since this is a community widget, I don't think it needs a lot of reviewing. One thing that would be nice to think about is how sturdy it is to the future. If a user has a fiddle made and saved in their doc, and this custom widget code is upgraded, how likely is everything to be ok. It might help to add a README to specify what is in the stored options, since that is the state that'll be persisted in people's documents and what the widget needs to remain compatible with. I see a `_js` and a `_html` field. `_html` is inserted basically verbatim within a template, and the `_js` is within a script tag afterwards, right? That might be all you need to say about the storage format and how it is interpreted.

Reply:
I added the README and the updated help page.
Unrelated change is the new `grist:serve` script in package.json that spins up grist container that points to the development server, it makes developing widget faster. I even considered replacing the dev script with it.